internal/services/recoveryservices/site_recovery_replication_recovery_plan_resource.go

```diff
@@ -0,0 +1,542 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type SiteRecoveryReplicationRecoveryPlanModel struct {
+	Name                   string               `tfschema:"name"`
+	RecoveryGroup          []RecoveryGroupModel `tfschema:"recovery_group"`
+	RecoveryVaultName      string               `tfschema:"recovery_vault_name"`
+	ResourceGroupName      string               `tfschema:"resource_group_name"`
+	SourceRecoveryFabricId string               `tfschema:"source_recovery_fabric_id"`
+	TargetRecoveryFabricId string               `tfschema:"target_recovery_fabric_id"`
+}
+
+type RecoveryGroupModel struct {
+	GroupType                replicationrecoveryplans.RecoveryPlanGroupType `tfschema:"group_type"`
+	PostAction               []ActionModel                                  `tfschema:"post_action"`
+	PreAction                []ActionModel                                  `tfschema:"pre_action"`
+	ReplicatedProtectedItems []string                                       `tfschema:"replicated_protected_items"`
+}
+
+type ActionModel struct {
+	ActionDetailType        string                                              `tfschema:"action_detail_type"`
+	FabricLocation          replicationrecoveryplans.RecoveryPlanActionLocation `tfschema:"fabric_location"`
+	FailOverDirections      []string                                            `tfschema:"fail_over_directions"`
+	FailOverTypes           []string                                            `tfschema:"fail_over_types"`
+	ManualActionInstruction string                                              `tfschema:"manual_action_instruction"`
+	Name                    string                                              `tfschema:"name"`
+	RunbookId               string                                              `tfschema:"runbook_id"`
+	ScriptPath              string                                              `tfschema:"script_path"`
+}
+
+type SiteRecoveryReplicationRecoveryPlanResource struct{}
+
+var _ sdk.ResourceWithUpdate = SiteRecoveryReplicationRecoveryPlanResource{}
+
+func (r SiteRecoveryReplicationRecoveryPlanResource) ResourceType() string {
+	return "azurerm_site_recovery_replication_recovery_plan"
+}
+
+func (r SiteRecoveryReplicationRecoveryPlanResource) ModelObject() interface{} {
+	return &SiteRecoveryReplicationRecoveryPlanModel{}
+}
+
+func (r SiteRecoveryReplicationRecoveryPlanResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return replicationrecoveryplans.ValidateReplicationRecoveryPlanID
+}
+
+func (r SiteRecoveryReplicationRecoveryPlanResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"recovery_vault_name": {
```

Comment:
If it can be got by `azurerm_recovery_services_vault.id`, recommend replacing with `recovery_vault_id` like existing resources.